src/content.js

```diff
@@ -90,6 +91,78 @@ function content(content_id, callback) {
   });
 }
 
+// Now that a content document is available, perform post-processing calls in parallel.
+function postprocess(presented_url, content_doc, callback) {
+  async.parallel([
+    async.apply(related, content_doc),
+    async.apply(layout, presented_url, content_doc)
+  ], function (err, output) {
+    if (err) return callback(err);
+
+    var output_doc = {
+      envelope: content_doc.envelope,
+      results: output[0],
+      layout: output[1]
+    };
+
+    callback(null, output_doc);
+  });
+}
+
+// If the content document contains any query results, resolve their content IDs to presented URLs.
+function related(content_doc, callback) {
+  if (! content_doc.results) {
+    return callback(null, {});
+  }
+
+  var resultSetNames = _.keys(content_doc.results);
```

Comment:
I still need to do a camelCasing pass on the presenter, to be consistent with the content service...